### PR TITLE
Look for shared baserom directory using XDG_DATA_DIR instead of polluting the home directory

### PR DESCRIPTION
**Pull request by mpharoah**
A long time ago, people used to put all their application data in hidden directories in the home directory. This was very annoying because you would end up with tons of folders in your home directory. And so, the basedir-spec freedesktop specification was made to define where applications should store their user data, and everyone switched to using that. HackerSM64 should do the same.

The specification: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
Simplification from the Arch Linux wiki: https://wiki.archlinux.org/title/XDG_Base_Directory

tl;dr: don't pollute the home directory. Look for the baserom (and any other future things we want to persist across checkouts) in `$XDG_DATA_HOME/HackerSM64` (fallback to `~/.local/share/HackerSM64` if `XDG_DATA_HOME` is empty or not defined) instead of in a new directory in home.

### CODE REVIEW

**Review by gheskett — APPROVED** [review]
@someone2639 Please make sure this is compatible with your changes.

**Review by someone2639 — APPROVED** [review]
